src/client/src/client/cluster.rs

```diff
@@ -71,23 +72,14 @@ impl Fluvio {
         let (socket, config, versions) = inner_client.split();
         let socket = AllMultiplexerSocket::shared(socket);
 
-        let spu_pool = OnceCell::new();
-        Ok(Self {
+        let spu_pool = Arc::new(SpuPool::start(config.clone(), &socket).await?);
+        let inner = Arc::new(FluvioInner {
```

Comment:
Since `Fluvio` holds `FluvioInner` in an Arc, we probably do not need `Arc<SpuPool>` inside of that as well

Reply:
Other than that everything looks good

Reply:
I realize that may not have been clear. Instead of `Arc<SpuPool>` in `FluvioInner` we can have just `SpuPool`

Reply:
:+1: